hoodie-client/pom.xml

```diff
@@ -78,6 +79,71 @@
       <artifactId>hoodie-timeline-service</artifactId>
       <version>${project.version}</version>
     </dependency>
+
+    <!-- Logging -->
+    <dependency>
+      <groupId>log4j</groupId>
+      <artifactId>log4j</artifactId>
+    </dependency>
+
+    <!-- Parquet -->
+    <dependency>
+      <groupId>org.apache.parquet</groupId>
+      <artifactId>parquet-avro</artifactId>
+    </dependency>
+    <dependency>
+      <groupId>org.apache.parquet</groupId>
+      <artifactId>parquet-hadoop</artifactId>
+    </dependency>
+
+    <!-- Spark -->
+    <dependency>
+      <groupId>org.apache.spark</groupId>
+      <artifactId>spark-core_2.11</artifactId>
+    </dependency>
+    <dependency>
+      <groupId>org.apache.spark</groupId>
+      <artifactId>spark-sql_2.11</artifactId>
+    </dependency>
+
+    <!-- Dropwizard Metrics -->
+    <dependency>
+      <groupId>io.dropwizard.metrics</groupId>
+      <artifactId>metrics-graphite</artifactId>
+    </dependency>
+    <dependency>
+      <groupId>io.dropwizard.metrics</groupId>
+      <artifactId>metrics-core</artifactId>
+    </dependency>
+
+    <dependency>
+      <groupId>com.google.guava</groupId>
+      <artifactId>guava</artifactId>
+    </dependency>
+
+    <dependency>
+      <groupId>com.beust</groupId>
+      <artifactId>jcommander</artifactId>
+      <version>1.48</version>
+    </dependency>
+
+    <dependency>
+      <groupId>org.htrace</groupId>
+      <artifactId>htrace-core</artifactId>
+      <version>3.0.4</version>
+    </dependency>
+
+    <!-- Hadoop -->
+    <dependency>
```

Comment:
should hadoop be higher up on the list.. after spark? any reasons for having this here specifically?

Reply:
My reasoning for placing Hadoop far down the classpath, is that it brings many dependencies with it, and given we build against very old versions of Hadoop, these older dependencies will often shade newer ones if placed near the top.

My goal with this PR is to make it easier to upgrade things like avro, parquet, spark, etc.
Which means I want the versions specified in the base pom to be the version in every jar.

Reply:
makes sense.. one thing to note is that., if  you concerned about bundles primarily as you upgrade these things, then hadoop/spark are in "provided" scope. so they will be excluded anyway.